torch/csrc/jit/script/compiler.h

```diff
@@ -17,6 +17,14 @@ struct CallsiteDescriptor {
   bool allow_varargs;
 };
 
+struct DefAndTypes {
+  DefAndTypes(Def def, std::vector<TypePtr> arg_types, TypePtr return_type)
+    : def(std::move(def)), arg_types(arg_types), return_type(return_type) {}
+  Def def;
+  std::vector<TypePtr> arg_types;
```

Comment:
We should use a `FunctionSchema` object here instead of raw types because it is the right conceptual object for the information we are recording.